app/controllers/entry.rb

```diff
@@ -0,0 +1,73 @@
+include FileUtils::Verbose
+require 'fileutils'
+
+
+#create
+get '/upload' do
+  @cohorts    = Cohort.all
+  @languages  = Language.all
+  @phases     = Phase.all
+  erb :'entry/upload'
+end
+
+post '/upload' do
+  user_id     = current_user_id
+  phase_id    = Phase.find_by_name(params[:phase]).id
+  cohort_id   = Cohort.find_by_name(params[:cohort]).id
+  language_id = Language.find_by_name(params[:language]).id
+
+  if params[:file]
+    entry = Entry.create(params[:entry])
+    entry.update_attributes(phase_id:    phase_id, \
+                            cohort_id:   cohort_id, \
+                            user_id:     user_id \
+                           )
+
+    Coding.create(entry_id: entry.id, language_id: language_id)
+
+    FileUtils::mkdir_p "public/uploads/#{entry.id}"
+
+    @tempfile = params[:file][:tempfile]
+    @filename = params[:file][:filename]
+    cp(@tempfile.path, "public/uploads/#{entry.id}/#{@filename}")
+
+    redirect("/entry/#{entry.id}")
```

Comment:
This is getting a bit big for controller actions. I think it would be cleaner if you moved the file magic out of here and into a module or model area.